deku-derive/src/macros/deku_read.rs

```diff
@@ -352,7 +362,36 @@ fn emit_enum(input: &DekuData) -> Result<TokenStream, syn::Error> {
         });
     }
 
-    // println!("{}", tokens.to_string());
+    // create `deku_id` function for retrieving at runtime the id value
+    let deku_id_id_type = if id_type.is_some() || input.ctx.is_none() {
+        id_type.unwrap()
```

Comment:
this could panic if `id_type==None` and `input.ctx==Some`

Would perfer an `if let` binding

Reply:
Fixed